### PR TITLE
Add support for Ansible 2.4

### PR DESCRIPTION
**Pull request by mixja**
Adds support for Ansible 2.4.

Note these are breaking changes.

### CODE REVIEW

**Comment by mixja** [comment]
Successfully tested with https://github.com/Casecommons/intake-api-aws/commit/74dd742b1d8c1d0d5ec9cf41f58570e6c627e690

**Comment by pemacc** [comment]
Looks good. @christian-casecommons any reason why I should not merge this pull request to master banch.

**Comment by christian-casecommons** [comment]
None, just as long as we are still good on ~v2.3

File: README.md [hunk 1]
```diff
@@ -268,6 +268,10 @@ The following is an example of a playbook configured to use this role.  Note the
 
 ## Release Notes
 
+### Version 2.4.0
+
+- **BREAKING CHANGE** : Compatibility fixes for Ansible 2.4.x
```
Comment (christian-casecommons) [inline README.md]: Ahh, I guess there is - @mixja will this break 2.3 runners?

Comment (mixja) [reply README.md]: Yes this will break with Ansible 2.3

**Comment by mixja** [comment]
@pemacc @christian-casecommons - before merging this to master you need to consider if you will support < Ansible 2.3 going forward.  With this merged, although you can still use < Ansible 2.4 with earlier versions, you won't be able to introduce any changes that are compatible with Ansible 2.3.

**Comment by christian-casecommons** [comment]
Hey @mixja, I am currently running through our entire stack (of stacks), so I'll go ahead try with ansible 2.4 and document/update the incompatibilities.

**Pull request merged**